### ISSUE TITLE
[Discussion] redesign Start page

### ISSUE THREAD

**Issue by RussKie** [gitextensions/gitextensions#3658]
The current start page (the dashboard) is not ergonomic, not user friendly and non intuitive.
I've been using GE for about 4 years and I always struggled with lack of scroll for repositories list, if you have many of them. Also up until I've read #3648 I had no idea there were categories. And not until I browsed through the code of the dashboard control I saw that you could add RSS feeds...
And on top of that - the page is slow and inefficient (I timed it and it takes 1.5-1.8 sec to render).

I'm sure GE can offer a better start page. For example, take an inspiration from VS and make something similar.
I'd like to volunteer to have a go and make a better start page, however I'd like to get some clarity from those in the know about the required functionality. Or anyone who actually uses the start page.

* RSS feeds
  - Does anyone uses it? 
  - If people use it, do they use more than one feed?
* Categories
  - What are they used for? What would be a use case for them?
  - Can they be substituted with "pinned" or "favourite" feature?

-----

Personally, in both GE and VS I use their start pages to do the following:
- _clone/start project_
- _open project (browse)_
- _open a recent project_

In VS I _pin projects_ to prevent them from disappearing.
Add to the list - "_remove project from the list_" or "_open containing folder_", I don't use these but these are probably handy to have.

So naturally as a user I would want to have these and see them as ergonomic and intuitive as possible.

**If we presume for a moment that GE doesn't have a start page, what else would be handy to have at the start page?** 

This may address/affect/obsolete #633, #1062, #1358, #1382, #1970,  #2228, #2328, #2905, #3083, #3220, #3624, #3648,  #3932

**Comment by mikecasas** [comment]
Would love to be able to load some kind of file for all your repos to the dashboard.

category services
   project 1
   project 2
category websites
  project 3
  project 4

**Comment by pmiossec** [comment]
I don't use rss feed (and don't really see the point of such feature). 

I didn't know about the category feature (and still don't understand what it is. I don't have an access to gitextensions at the moment to look at it...) 

I agree that this page could be improved a little by making some features easily and quickly usable for the user. The actual features provided (rss and categories) are too hard to discover :-( Perhaps add links to be able to directly access the settings from the dashboard. 

What I see is something similar to what you propose... 

In the dashboard, I would like to see favourites repositories and add a new one with 1 click from the recent list. 

I also think that this list should be the same than the one proposed in the repositories combo in the menu bar. 

So :+1: for the refactoring, even if I'm not a big user of the dashboard (but perhaps because of the not so good ergonomic... )

**Comment by mikecasas** [comment]
The category feature is just that, a way to categorize or group your repos. You may have a category of front end or scripts or personal projects.

**Comment by RussKie** [comment]
Here's an idea

Windowed, 1268x843
![image](https://cloud.githubusercontent.com/assets/4403806/25310627/65812bf6-282c-11e7-98f1-bcb578c6faf2.png)

full screen, 1900x1200
![image](https://cloud.githubusercontent.com/assets/4403806/25310642/a87dc63a-282c-11e7-9004-4abf0f52d449.png)

----

I put "Change log" on the right, as it visually calls for something to be there... It doesn't have to a be a changelog, but I couldn't come up with an idea what to put there...
![image](https://cloud.githubusercontent.com/assets/4403806/25310661/091ac060-282d-11e7-8ee5-3052d8b1ad7d.png)

The color scheme is as well up for a discussion, but I personally wouldn't want it to be plain white, it feels too blend and too unfinished. And frighting, if you wish, like a blank piece of paper :) 



I haven't forgotten about categories, it is something in works.
Right now wishing to gauge the general feel towards the draft.

**Comment by jbialobr** [comment]
>The color scheme is as well up for a discussion.

Just make sure it works properly with different color schemes. #3630, #711.

**Comment by RussKie** [comment]
Good point

**Comment by RussKie** [comment]
dark theme
<img width="1008" alt="screen shot 2017-04-27 at 9 22 40 pm" src="https://cloud.githubusercontent.com/assets/4403806/25481366/2d7905b4-2b90-11e7-8035-7244d5b36e57.png">

NB: the color palette is hand crafted, not using the Windows colors.

**Comment by vbjay** [comment]
The dark theme is freaking awesome.

**Comment by RussKie** [comment]
Thanks :)
I can't work out how to set bloody W10 to dark theme though

**Comment by vbjay** [comment]
[image: image.png]

**Comment by vbjay** [comment]
![image](https://cloud.githubusercontent.com/assets/3629489/25482456/e6940524-2b1f-11e7-83a3-b15556b370e5.png)

**Comment by RussKie** [comment]
this doesn't work
this affects metro apps and has no effect on desktop apps

<img width="751" alt="screen shot 2017-04-27 at 10 55 36 pm" src="https://cloud.githubusercontent.com/assets/4403806/25484265/ad20bc10-2b9c-11e7-8017-3bdc104f3bf1.png">

**Comment by RussKie** [comment]
OT: got w10 theming working with @mwerle's help #3630

**Comment by RussKie** [comment]
Few fresh screenshots:

Got categories working:
<img width="1022" alt="screen shot 2017-05-04 at 12 02 13 am" src="https://cloud.githubusercontent.com/assets/4403806/25664278/7fe4b2fe-305d-11e7-8be7-712f47d20c05.png">

<img width="1026" alt="screen shot 2017-05-04 at 12 02 32 am" src="https://cloud.githubusercontent.com/assets/4403806/25664313/91ffe850-305d-11e7-883f-046c83b85b58.png">

<img width="1021" alt="screen shot 2017-05-04 at 12 03 34 am" src="https://cloud.githubusercontent.com/assets/4403806/25664327/9d4abff0-305d-11e7-905e-39fd88fd3d4f.png">

When changing color themes two thing can't be re-styled - scrollbars and the dropdown
![parallels picture](https://cloud.githubusercontent.com/assets/4403806/25664480/0b5eb6ea-305e-11e7-82be-4f7f6a549151.png)

----

I'm still looking for ideas on how to occupy the emptiness on the right...
![screen shot 2017-05-04 at 12 24 12 am](https://cloud.githubusercontent.com/assets/4403806/25665269/7001bfaa-3060-11e7-8511-3a15deba2d15.png)

**Comment by mwerle** [comment]
What about people not using widescreen monitors and/or not maximising applications?

Basically what I'm saying is, make that "right-hand-emptiness-filler" optional, and have it auto-hide if the window is not wide enough for it,

**Comment by RussKie** [comment]
duly noted, and I was planning for something like that anyway.
the question still remains for users with wide monitors - I expect these users form the larger group of GE users.

**Comment by RussKie** [comment]
<img width="1328" alt="screen shot 2017-05-04 at 11 12 04 pm" src="https://cloud.githubusercontent.com/assets/4403806/25705040/31fb0d00-311f-11e7-8133-1d40b1407e8a.png">
<img width="1332" alt="screen shot 2017-05-04 at 11 11 14 pm" src="https://cloud.githubusercontent.com/assets/4403806/25705044/33f6b460-311f-11e7-9b4c-10a10477a13e.png">

**Comment by francoas** [comment]
For me categories are very important
I have dozens of repositories I am working on, divided in categories

I would like to be able to see all of them in a MULTI COLUMN page rather then scrolling up and down 

Here is an example

![image](https://cloud.githubusercontent.com/assets/3007941/25889426/03932c28-356a-11e7-88fa-6391172c5289.png)

**Comment by RussKie** [comment]
@francoas your desired design layout may have limited use in general, because there is no path information. If you take paths in consideration it may not be able to fit as much information on the screen (unless you have a huge screen of course).

No matter how interface changes there will be happy and unhappy users. It is impossible to cater for everyone's needs, the best we can do is to strike a middle ground and offer something which works for the majority.

My proposal is to have a category dropdown:
 
![image](https://cloud.githubusercontent.com/assets/4403806/25779590/c356510a-335f-11e7-92ea-1fed96ecf4f7.png)

![image](https://cloud.githubusercontent.com/assets/4403806/25779597/db5f4108-335f-11e7-98e7-0868afe83b74.png)

There are more screenshots in #3693

**Comment by jbialobr** [comment]
>If you take paths in consideration it may not be able to fit as much information on the screen (unless you have a huge screen of course).

Perhaps the Shortening strategy should be applied.

**Comment by francoas** [comment]
@jbialobr 

> Perhaps the Shortening strategy should be applied.

Yeah... and, besides, widgets like wpf wrap... automatically adjiust numbers of columns according to space
Or the user can configure the number of colums and then shortening applies: default = 1 => current behaviour. This way everybody can be happy

Anyway... also a collapsable category (much easier I guess) could do.
See picture

![image](https://cloud.githubusercontent.com/assets/3007941/25998019/584e8706-371f-11e7-9ed4-4231712e47e2.png)

And yes... I have a large display :)

Best regards

**Comment by jbialobr** [comment]
>Also up until I've read #3648 I had no idea there were categories.

So there are users who use categories. Moreover they put an effort to organize them. I think we can not deprecated them. Recent repos can be cleared, but it should not affect the repos pinned to the dashboard.

**Comment by RussKie** [comment]
The core functionality is still there - there are means to categorise repositories, manage categories and select between them.

![image](https://cloud.githubusercontent.com/assets/4403806/26473927/7e652f58-41f1-11e7-9478-a77ba4e263ec.png)


I have missed the import of existing categorisations into the new structure (an oversight since I don't use categories), but it is something can be fixed.

**Comment by jbialobr** [comment]
Managing categories with combo is not handy. One can have few categories with few repos. When all these categories (or at least the most frequently used) fit on the dashboard then switching between categories is painful. 

>The core functionality is still there

Recent repos can be cleared, but it should not affect the repos pinned to categories.
The core functionality let the user to assign a repo to a category at a desired position. Categories can change their position on the dashboard. This allows user to have a stable dashboard structure.

**Comment by RussKie** [comment]
Thank you @jbialobr though I wish you made this known a little bit earlier, I really do...

It looks as though you take your contributors for granted. And I am not (just) talking about myself here. Up until recently you had PRs sitting in limbo for years. Most of them weren't even acknowledged by the project owners.
As an repo owner the very least you can do is to acknowledge and thank your contributors for their submissions, especially for those required significant time/effort. I personally know two individuals who won't be contributing to your project any more, citing neglect from the project owners and time it took for their PRs to be acknowledged and dealt with. Hell, it is faster to get a fix in to Microsoft repos than in here...

Also it is incorrect to assume or presume that you, as the project owner, know how your software is *actually* used, what features users use, like or dislike. 
Objectively, as a user, I can tell you - Git Extensions is a wonderful tool packed with features. However for the most part the UI is outdated, bordering on ugly. Have non-default Windows settings or custom font? The UI becomes un-/barely usable. 
It also has a range of subtle features hidden away - as a result they are of little, if any, value to your users.
Subjectively, as a developer.... well I won't go there.


The fact, that I started this thread and then subsequently submitted a PR, indicate my ultimate displeasure with the current functionality offered by the dashboard. Unlike most users I chose to do something about it, instead of just submitting a bug "it does not work".
At the beginning of the thread I _stated the problem_, my _personal vision for the new implementation_ and **sought feedback**. I pretty much looked up to you - the repo owners - those who have the best knowledge of the underlying codebase and functionality it offers. Sadly you choose not to participate.
I spent a considerable amount time and effort building a new dashboard which works in different resolutions, color themes and with different font sizes. I posted a number of screenshots showing the functionality, again my efforts did not seem to engage your attention then...


What I have built may not satisfy everyone, and that is not (entirely) my goal. First and foremost I plan to satisfy *my* user needs, since I am the one working on the feature. Second come users, who I interact with as part of my job. Third - certain users, who have issues similar to mine, and who voiced their problems here. Everyone else comes last.

Speaking of which. How many users who actually use the dashboard and its features do you know? 
I know dozens of developers in the organisation I work for and outside of it, who use Git Extensions and who don't use dashboard at all, nor aware or care about "categories" feature. 
I could count less than half a dozen of users (e.g. @francoas, @Shepdog87, @webmaster33, @mikecasas) who use the current dashboard and functionality it offers. And even these users are not entirely happy with existing functionality.
Of course, there are heaps more users who use categories, but there are even more who don't use them at all.

For the record, I respect opinions and desires of above mentioned users, however I am unable to accommodate _all_ of their requirements for technical or practical reasons. Though I am sure we can reach a compromise and work from there.

**Comment by jbialobr** [comment]
I am sorry you feel this way - it was not my intention. I think you made a false assumption
of what you can expect from me. I would love to work on this project full time and have enough time to shower each issue with attention, but unfortunately, the same as you, I devote my private time (free of charge) to contribute to this project. I think you also misunderstood my role in this project. I am a member not the owner. I was asked to help maintaining the Git Extensions project - not to take the ownership. I am not in the project from the beginig and there was a long period of time I was not actively participating in the project. I try to help whenever I can and try to do my best but we did not make any agreement of what you can expect from me. 

>It looks as though you take your contributors for granted. And I am not (just) talking about myself here. Up until recently you had PRs sitting in limbo for years. Most of them weren't even acknowledged by the project owners.

You can't be farther from the truth. As you can see there is so much to do, the project lacks a maintainer workforce. There were periods when it lacked it even more.

>As an repo owner the very least you can do is to acknowledge and thank your contributors for their submissions, especially for those required significant time/effort. I personally know two individuals who won't be contributing to your project any more, citing neglect from the project owners and time it took for their PRs to be acknowledged and dealt with. Hell, it is faster to get a fix in to Microsoft repos than in here...

I understand your regret, but please try to understand others - not everything is just black or white, take time and effort to wonder why things happen the way they happened. Take into account others abilities and disabilities - not only assume a bad intention. 

>Also it is incorrect to assume or presume that you, as the project owner, know how your software is actually used, what features users use, like or dislike.

I do not assume that, but I take the responisiblity for changes I accept to be merged in.
I don't use the dashboard at all, so I didn't take a part in the discussion of this issue (resolving bugs and issues assigned to the 2.50 milestone seemed to me more important). Then comments apeared from users who use dashboard. I don't want that they loose the functionality they have and use. 

>Subjectively, as a developer.... well I won't go there.

Neither, will I.

>...Everyone else comes last.

We are not on the same page here. The latest version was downloaded 100K times.
Despite of 99K of the users don't participate in discussions they are not less important.
Again, try to wonder what can be the reasons they do not watch the GitExtensions proejct.
Every contributor wants to satisfy his/her needs but they should not come at expense of other users.

> Though I am sure we can reach a compromise and work from there.

So, do I.

**Comment by mikecasas** [comment]
@RussKie All open source developers/contributors battle the work-life balance and demands on our personal time.

For categories, I would be happy with just copying over some text file that has all my repos. The UI changes look great though.

**Comment by pmiossec** [comment]
>All open source developers/contributors battle the work-life balance and demands on our personal time.

@mikecasas yes but as open source maintainers, we are so much more user to this behaviour:


How to contribute to open source:
1. Fork the repo
2. Realize the problem is actually more complicated than you thought
3. Submit a bug

Got from https://twitter.com/pmiossec/status/870663692857339904

And I know for sure that sometimes, it's boring... ;-)

**Comment by mikecasas** [comment]
Did I miss the contributing page?

**Comment by RussKie** [comment]
I feel like I need to clarify few things.

First of all - my apologies to @jbialobr, my rant wasn't directed at him personally, but at the owners/members group. These people are ultimately the product owners, they bear the responsibility and decide what goes in or doesn't by the virtue of ability to merge (and that group includes @jbialobr ). 

The lack of leadership ultimately results in inability to innovate. At some point the product owners must make hard decisions whether to introduce or obsolete features and stay relevant, or let the project stagnate and die... It is impossible to keep everyone happy.
Also if there is no one around to explain what use-case for feature A is or to make a judgement call whether feature B is any longer useful, the project is doomed.
There is a lot of code in the codebase written almost 10 years ago. You've done a great job, but is what you've done back then still relevant or appropriate?
The whole world, the IT industry, development practices, UX guidelines, user expectations changed so much that a lot of things normal back then no longer work or make sense...

Look at any sizable software company - Microsoft, Google, Apple - all of them innovate and change, by dropping and introducing features, changing interfaces and our habits of how we all use their software and hardware. By doing so they often piss a lot of people off, but in the end of the day most of these changes are for the better (except Windows 8 😄 ).

Incidentally yesterday I talked to one of the graduate developers - he was evaluating [gitkraken](https://www.gitkraken.com/) and he was even considering buying it from his own pocket.
The main reason - it looked cool and modern unlike Git Extensions.

I am impartial to the project, I use it every day, and my main concern that if product owners (and we as the community) do no innovate and "reload" Git Extensions with modern, sleek and usable UI then you may find that in the next release there could be significantly less downloads, and then one day there wouldn't be anyone interested at all...

**Comment by RussKie** [comment]
Duplicating from #3693

Trying to address fundamental Mono issues with docked and scrollable controls I had an idea - to use a ListView with Tile view.

<img width="1013" alt="screen shot 2017-06-04 at 9 46 28 pm" src="https://cloud.githubusercontent.com/assets/4403806/26761262/53e24de4-496f-11e7-9a2a-6cea1db781b0.png">
<img width="1113" alt="screen shot 2017-06-04 at 9 46 11 pm" src="https://cloud.githubusercontent.com/assets/4403806/26761263/54d3c4f8-496f-11e7-8a72-1dc6b47059f4.png">

<img width="994" alt="screen shot 2017-06-04 at 10 16 54 pm" src="https://cloud.githubusercontent.com/assets/4403806/26761511/3630aef8-4974-11e7-8b93-57b94b915e43.png">
<img width="1075" alt="screen shot 2017-06-04 at 10 17 43 pm" src="https://cloud.githubusercontent.com/assets/4403806/26761512/37005c5c-4974-11e7-83f8-91fc45cb64b3.png">


I think it may even be possible to address other concerns - such as visualising categories on the same page, multi columns and shortening strategy.

**Comment by jbialobr** [comment]
@RussKie I like the idea of 'uncategorised' - it gives a clue that there exist catagories. 
I would consider the disappearing scroll bar on resize as a minor issue comparing to the existing mono issues.

Discussing `The lack of leadership` should be started as a separate issue to keep this issue on-topic.

**Comment by RussKie** [comment]
For further updates and discussions regarding the new dashboard please head to #3693

**Comment by klorinczi** [comment]
I might change the order of the two sections.
1) Favourite repos 
2) Recent repos

Reason: Favourite repos are the most used repos, so I need them on the first place.
Recent repos contains (should contain) only those repos, which are not added into Favourite repos .
